src/reductiveContext.re

```diff
@@ -0,0 +1,91 @@
+[@bs.config {jsx: 3}];
+module type Config = {
+  type state;
+  type action;
+
+  let store: Reductive.Store.t(action, state);
+};
+
+module Make = (Config: Config) => {
+  let storeContext = React.createContext(Config.store);
+
+  module ContextProvider = {
+    let make = React.Context.provider(storeContext);
+    let makeProps = (~value, ~children, ()) => {
+      "value": value,
+      "children": children,
+    };
+  };
+
+  module Provider = {
+    [@react.component]
+    let make = (~children) => {
+      <ContextProvider value=Config.store> children </ContextProvider>;
+    };
+  };
+
+  let useSelector = selector => {
```

Comment:
This function seems massively complex and I'm not really sure why. As far as I can tell we just want to memoize selector+storeFromContext and get the resulting view of state?

Would something like this work?

```reason
let storeFromContext = React.useContext(storeContext);
let stateFromContext = Reductive.Store.getState(storeFromContext);
React.useMemo2(
  () => selector(stateFromContext),
  (selector, stateFromContext)
);
```